File: python/mirascope/ops/exceptions.py
"""Mirascope Ops exception hierarchy for unified error handling across providers."""


class MirascopeOpsError(Exception):
    """Base exception for all Mirascope Ops errors."""

    original_exception: Exception | None


class ConfigurationError(MirascopeOpsError):
    """Raised when Ops configuration is invalid."""


class ClosureComputationError(MirascopeOpsError):
    """Raised when the closure for a function cannot be computed properly."""

    qualified_name: str

    def __init__(self, qualified_name: str) -> None:
        """Initializes an instance of `ClosureComputationError`."""
        self.qualified_name = qualified_name

File: typescript/src/ops/exceptions.ts
/**
 * Exception classes for the Mirascope ops module.
 */

import { MirascopeError } from "@/llm/exceptions";

/**
 * Raised when ops configuration is invalid.
 */
export class ConfigurationError extends MirascopeError {
  constructor(message: string) {
    super(message);
  }
}

/**
 * Raised when function closure cannot be computed.
 */
export class ClosureComputationError extends MirascopeError {
  readonly qualifiedName: string;

  constructor(qualifiedName: string, message: string) {
    super(message);
    this.qualifiedName = qualifiedName;
  }
}
